Remove excessive indirection with sending remote glyph drawing commands

kkinnunen-apple opened the pull request:
#### fcc4fa8ea686bf40d49530a432021102bf497ca9
<pre>
Remove excessive indirection with sending remote glyph drawing commands
<a href="https://bugs.webkit.org/show_bug.cgi?id=288635">https://bugs.webkit.org/show_bug.cgi?id=288635</a>
<a href="https://rdar.apple.com/145681066">rdar://145681066</a>

Reviewed by Simon Fraser.

DisplayList::Recorder interface would cause overly many virtual function
calls and would prevent backend-specific caching behavior.

Consider callgraph for `context-&gt;drawDecomposedGlyphs()`:
Recorder::drawDecomposedGlyphs() &lt;vfunc&gt;
    RemoteDisplayListRecorderProxy::recordResourceUse(font) &lt;vfunc&gt;
    RemoteDisplayListRecorderProxy::recordResourceUse(decomposedGlyphs) &lt;vfunc&gt;
    RemoteDisplayListRecorderProxy::recordDecomposedGlyphs() &lt;vfunc&gt;
        message send...

Instead, fix it in this patch to be:
RemoteDisplayListRecorderProxy::drawDecomposedGlyphs() &lt;vfunc&gt;
    RemoteDisplayListRecorderProxy::recordResourceUse(font) &lt;normal&gt;
    RemoteDisplayListRecorderProxy::recordResourceUse(decomposedGlyphs) &lt;normal&gt;
    message send..

In addition, fixes in-process display lists to not have redundant resource
hash map access for font and decomposed glyphs. Previously, display list
items would store the identifier, and the display list resource cache
hash map would store the identifier -&gt; Ref&lt;Font&gt; and
identifier -&gt; Ref&lt;DecomposedGlyphs&gt;. The hash map lookup is unneeded,
the items should store the ref directly.

In future patches, this allows recordResourceUse calls to record and return
backend command specific data without the need for this data to leak
into the WebCore. For example, current identifiers should not be defined
in WebCore at all, but they are, since recordResourceUse is there and
display list resource cache redundantly uses those.

This is continuation of 284963@main.

Remove the pessimizing PositionedGlyphs struct. It was only used to
serialize a message parameter, it did no payload work. Instead, it
incurred a copy of glyphs and advances, because PositionedGlyphs was
redundantly used in hold sites to hold the vectors.

Move DrawGlyphsRecorder to CoreText. It was only implemented in
CoreText and had zero cross-platform capable code. It is also used in
a hacky way for Cocoa: it is used to do DrawGlyphs decomposition as
well as a way to convert CG drawText into GraphicsContext drawText
for attachment rendering. This structure imposes hacks to the overall
system that have no relevancy to the other ports.

* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-color-expected.txt:
* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-shadow-unshared.html:
* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/FreeType.cmake:
* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
(WebCore::DecomposedGlyphs::DecomposedGlyphs):
* Source/WebCore/platform/graphics/DecomposedGlyphs.h:
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::displayListForTextRun const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawGlyphsImmediate):
(WebCore::GraphicsContext::drawDecomposedGlyphs):
* Source/WebCore/platform/graphics/GraphicsContext.h:
(WebCore::GraphicsContext::drawGlyphsAndCacheResources): Deleted.
* Source/WebCore/platform/graphics/PositionedGlyphs.cpp: Removed.
* Source/WebCore/platform/graphics/PositionedGlyphs.h: Removed.
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.cpp: Renamed from Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorderCoreText.cpp.
(WebCore::DrawGlyphsRecorder::DrawGlyphsRecorder):
(WebCore::DrawGlyphsRecorder::recordDrawGlyphs):
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.h: Renamed from Source/WebCore/platform/graphics/DrawGlyphsRecorder.h.
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp:
(WebCore::DisplayList::applyItem):
(WebCore::DisplayList::applyDrawGlyphs): Deleted.
(WebCore::DisplayList::applyDrawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::apply const):
(WebCore::DisplayList::DrawGlyphs::dump const):
(WebCore::DisplayList::DrawDecomposedGlyphs::apply const):
(WebCore::DisplayList::DrawDecomposedGlyphs::dump const):
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
(WebCore::DisplayList::DrawGlyphs::font const):
(WebCore::DisplayList::DrawGlyphs::glyphs const):
(WebCore::DisplayList::DrawGlyphs::advances const):
(WebCore::DisplayList::DrawGlyphs::localAnchor const):
(WebCore::DisplayList::DrawGlyphs::fontSmoothingMode const):
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs):
(WebCore::DisplayList::DrawDecomposedGlyphs::font const):
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphs const):
(WebCore::DisplayList::DrawGlyphs::fontIdentifier const): Deleted.
(WebCore::DisplayList::DrawGlyphs::positionedGlyphs const): Deleted.
(WebCore::DisplayList::DrawGlyphs::anchorPoint const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::fontIdentifier const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphsIdentifier const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::Recorder):
(WebCore::DisplayList::Recorder::drawGlyphs):
(WebCore::DisplayList::Recorder::shouldDeconstructDrawGlyphs const): Deleted.
(WebCore::DisplayList::Recorder::drawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::Recorder::drawGlyphsAndCacheResources): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.h:
(WebCore::DisplayList::Recorder::drawGlyphsMode const):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::drawGlyphsImmediate):
(WebCore::DisplayList::RecorderImpl::drawDecomposedGlyphs):
(WebCore::DisplayList::RecorderImpl::recordDrawGlyphs): Deleted.
(WebCore::DisplayList::RecorderImpl::recordDrawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
* Source/WebCore/platform/graphics/harfbuzz/DrawGlyphsRecorderHarfBuzz.cpp: Removed.
* Source/WebCore/platform/graphics/skia/DrawGlyphsRecorderSkia.cpp: Removed.
* Source/WebCore/platform/graphics/win/DrawGlyphsRecorderWin.cpp: Removed.
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintAttachmentText):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::cacheDecomposedGlyphs):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_cannot_be_forward_declared):
(headers_for_type):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::RemoteDisplayListRecorderProxy):
(WebKit::RemoteDisplayListRecorderProxy::drawGlyphsImmediate):
(WebKit::RemoteDisplayListRecorderProxy::drawDecomposedGlyphs):
(WebKit::RemoteDisplayListRecorderProxy::recordDrawGlyphs): Deleted.
(WebKit::RemoteDisplayListRecorderProxy::recordDrawDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::cacheDecomposedGlyphs):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Tools/TestWebKitAPI/Tests/WebCore/cg/BifurcatedGraphicsContextTestsCG.cpp:
(TestWebKitAPI::TEST(BifurcatedGraphicsContextTests, Text)):

Canonical link: <a href="https://commits.webkit.org/292642@main">https://commits.webkit.org/292642@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/485bc3a65598ce19f871ae3fd811013b86bd9621

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/96665 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/16279 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/6403 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/101738 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/47185 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/98710 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/16575 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/24716 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/73673 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30893 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/99668 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/12473 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/87421 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54008 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/96152 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/12231 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/5221 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/46513 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/82333 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/5309 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/103761 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/23733 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/17304 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/82722 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/23983 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/83474 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82105 "Found 3 new API test failures: /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print-errors, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/print, /WebKitGTK/TestPrinting:/webkit/WebKitPrintOperation/close-after-print (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20609 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/26751 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/4282 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/17206 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/23695 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/28850 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/23354 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/26834 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/25095 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
```diff
@@ -6174,12 +6169,15 @@ struct WebCore::VideoFrameTimeMetadata {
-    WebCore::FontSmoothingMode smoothingMode;
+#if ENABLE(TOUCH_EVENT_REGIONS)
+header: <WebCore/EventRegion.h>
+[CustomHeader] struct WebCore::TouchEventListenerRegion {
```
Not sure why this shows up here now?

smfr commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -641,9 +641,9 @@
-		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24C2D6D2A4E00C63915 /* NetworkingExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2482D6D295400C63915 /* NetworkingExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Seems unrelated.

kkinnunen-apple committed:
Remove excessive indirection with sending remote glyph drawing commands
https://bugs.webkit.org/show_bug.cgi?id=288635
rdar://145681066

Reviewed by Simon Fraser.

DisplayList::Recorder interface would cause overly many virtual function
calls and would prevent backend-specific caching behavior.

Consider callgraph for `context->drawDecomposedGlyphs()`:
Recorder::drawDecomposedGlyphs() <vfunc>
    RemoteDisplayListRecorderProxy::recordResourceUse(font) <vfunc>
    RemoteDisplayListRecorderProxy::recordResourceUse(decomposedGlyphs) <vfunc>
    RemoteDisplayListRecorderProxy::recordDecomposedGlyphs() <vfunc>
        message send...

Instead, fix it in this patch to be:
RemoteDisplayListRecorderProxy::drawDecomposedGlyphs() <vfunc>
    RemoteDisplayListRecorderProxy::recordResourceUse(font) <normal>
    RemoteDisplayListRecorderProxy::recordResourceUse(decomposedGlyphs) <normal>
    message send..

In addition, fixes in-process display lists to not have redundant resource
hash map access for font and decomposed glyphs. Previously, display list
items would store the identifier, and the display list resource cache
hash map would store the identifier -> Ref<Font> and
identifier -> Ref<DecomposedGlyphs>. The hash map lookup is unneeded,
the items should store the ref directly.

In future patches, this allows recordResourceUse calls to record and return
backend command specific data without the need for this data to leak
into the WebCore. For example, current identifiers should not be defined
in WebCore at all, but they are, since recordResourceUse is there and
display list resource cache redundantly uses those.

This is continuation of 284963@main.

Remove the pessimizing PositionedGlyphs struct. It was only used to
serialize a message parameter, it did no payload work. Instead, it
incurred a copy of glyphs and advances, because PositionedGlyphs was
redundantly used in hold sites to hold the vectors.

Move DrawGlyphsRecorder to CoreText. It was only implemented in
CoreText and had zero cross-platform capable code. It is also used in
a hacky way for Cocoa: it is used to do DrawGlyphs decomposition as
well as a way to convert CG drawText into GraphicsContext drawText
for attachment rendering. This structure imposes hacks to the overall
system that have no relevancy to the other ports.

* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-color-expected.txt:
* LayoutTests/fast/text/glyph-display-lists/glyph-display-list-shadow-unshared.html:
* Source/WebCore/Headers.cmake:
* Source/WebCore/PlatformWin.cmake:
* Source/WebCore/SaferCPPExpectations/ForwardDeclCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/NoUncountedMemberCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/FreeType.cmake:
* Source/WebCore/platform/SourcesSkia.txt:
* Source/WebCore/platform/graphics/DecomposedGlyphs.cpp:
(WebCore::DecomposedGlyphs::create):
(WebCore::DecomposedGlyphs::DecomposedGlyphs):
* Source/WebCore/platform/graphics/DecomposedGlyphs.h:
* Source/WebCore/platform/graphics/FontCascade.cpp:
(WebCore::FontCascade::displayListForTextRun const):
* Source/WebCore/platform/graphics/GraphicsContext.cpp:
(WebCore::GraphicsContext::drawGlyphsImmediate):
(WebCore::GraphicsContext::drawDecomposedGlyphs):
* Source/WebCore/platform/graphics/GraphicsContext.h:
(WebCore::GraphicsContext::drawGlyphsAndCacheResources): Deleted.
* Source/WebCore/platform/graphics/PositionedGlyphs.cpp: Removed.
* Source/WebCore/platform/graphics/PositionedGlyphs.h: Removed.
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.cpp: Renamed from Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorderCoreText.cpp.
(WebCore::DrawGlyphsRecorder::DrawGlyphsRecorder):
(WebCore::DrawGlyphsRecorder::recordDrawGlyphs):
* Source/WebCore/platform/graphics/coretext/DrawGlyphsRecorder.h: Renamed from Source/WebCore/platform/graphics/DrawGlyphsRecorder.h.
* Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp:
(WebCore::DisplayList::applyItem):
(WebCore::DisplayList::applyDrawGlyphs): Deleted.
(WebCore::DisplayList::applyDrawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
(WebCore::DisplayList::DrawGlyphs::apply const):
(WebCore::DisplayList::DrawGlyphs::dump const):
(WebCore::DisplayList::DrawDecomposedGlyphs::apply const):
(WebCore::DisplayList::DrawDecomposedGlyphs::dump const):
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.h:
(WebCore::DisplayList::DrawGlyphs::DrawGlyphs):
(WebCore::DisplayList::DrawGlyphs::font const):
(WebCore::DisplayList::DrawGlyphs::glyphs const):
(WebCore::DisplayList::DrawGlyphs::advances const):
(WebCore::DisplayList::DrawGlyphs::localAnchor const):
(WebCore::DisplayList::DrawGlyphs::fontSmoothingMode const):
(WebCore::DisplayList::DrawDecomposedGlyphs::DrawDecomposedGlyphs):
(WebCore::DisplayList::DrawDecomposedGlyphs::font const):
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphs const):
(WebCore::DisplayList::DrawGlyphs::fontIdentifier const): Deleted.
(WebCore::DisplayList::DrawGlyphs::positionedGlyphs const): Deleted.
(WebCore::DisplayList::DrawGlyphs::anchorPoint const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::fontIdentifier const): Deleted.
(WebCore::DisplayList::DrawDecomposedGlyphs::decomposedGlyphsIdentifier const): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.cpp:
(WebCore::DisplayList::Recorder::Recorder):
(WebCore::DisplayList::Recorder::drawGlyphs):
(WebCore::DisplayList::Recorder::shouldDeconstructDrawGlyphs const): Deleted.
(WebCore::DisplayList::Recorder::drawDecomposedGlyphs): Deleted.
(WebCore::DisplayList::Recorder::drawGlyphsAndCacheResources): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorder.h:
(WebCore::DisplayList::Recorder::drawGlyphsMode const):
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
(WebCore::DisplayList::RecorderImpl::drawGlyphsImmediate):
(WebCore::DisplayList::RecorderImpl::drawDecomposedGlyphs):
(WebCore::DisplayList::RecorderImpl::recordDrawGlyphs): Deleted.
(WebCore::DisplayList::RecorderImpl::recordDrawDecomposedGlyphs): Deleted.
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.h:
* Source/WebCore/platform/graphics/harfbuzz/DrawGlyphsRecorderHarfBuzz.cpp: Removed.
* Source/WebCore/platform/graphics/skia/DrawGlyphsRecorderSkia.cpp: Removed.
* Source/WebCore/platform/graphics/win/DrawGlyphsRecorderWin.cpp: Removed.
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintAttachmentText):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp:
(WebKit::RemoteRenderingBackend::cacheDecomposedGlyphs):
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.h:
* Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(types_that_cannot_be_forward_declared):
(headers_for_type):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp:
(WebKit::RemoteDisplayListRecorderProxy::RemoteDisplayListRecorderProxy):
(WebKit::RemoteDisplayListRecorderProxy::drawGlyphsImmediate):
(WebKit::RemoteDisplayListRecorderProxy::drawDecomposedGlyphs):
(WebKit::RemoteDisplayListRecorderProxy::recordDrawGlyphs): Deleted.
(WebKit::RemoteDisplayListRecorderProxy::recordDrawDecomposedGlyphs): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::cacheDecomposedGlyphs):
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.h:
* Tools/TestWebKitAPI/Tests/WebCore/cg/BifurcatedGraphicsContextTestsCG.cpp:
(TestWebKitAPI::TEST(BifurcatedGraphicsContextTests, Text)):

Canonical link: https://commits.webkit.org/292642@main
